### ISSUE TITLE
Bad formatting of constraints in CREATE TABLE statements

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#7918]
Formatting this SQL:

```sql
create table x (
  i int,
  j int, 
  constraint c check (i > 0),
  primary key (i), unique (j), constraint c1 unique (i, j)
)
```

Yields:

```sql
create table x(
  i number null,
  j number null,
  constraint c
    check (i > 0),
    primary key (i), -- indented too much, here
  unique (j),
constraint c1 -- indented too little, here
  unique (
    i, 
    j
  )
)
```

**Comment by lukaseder** [comment]
The regression was introduced with #7409, https://github.com/jOOQ/jOOQ/commit/f5344c4bcfbd571e086fb6862520b3f76720879d

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.12. Backport scheduled for 3.11.6 (#7919)

**Comment by lukaseder** [comment]
We definitely need better automated regression tests for formatted SQL...
